### PR TITLE
[Merged by Bors] - feat: port Order.Cover

### PR DESCRIPTION
**Pull request by riccardobrasca**

### CODE REVIEW

File: Mathlib/Order/Cover.lean [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+variable [Preorder α] [Preorder β] {a b c : α}
+
+/-- `Wcovby a b` means that `a = b` or `b` covers `a`.
+This means that `a ≤ b` and there is no element in between.
+-/
+def Wcovby (a b : α) : Prop :=
+  a ≤ b ∧ ∀ ⦃c⦄, a < c → ¬c < b
+#align wcovby Wcovby
+
+-- mathport name: «expr ⩿ »
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:46]:
```suggestion
```

File: Mathlib/Order/Cover.lean [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+  ⟨fun h2 => h2.of_image f, fun hab => hab.image f h⟩
+#align set.ord_connected.apply_wcovby_apply_iff Set.OrdConnected.apply_wcovby_apply_iff
+
+@[simp]
+theorem apply_wcovby_apply_iff {E : Type _} [OrderIsoClass E α β] (e : E) : e a ⩿ e b ↔ a ⩿ b :=
+  (OrdConnected_range (e : α ≃o β)).apply_wcovby_apply_iff ((e : α ≃o β) : α ↪o β)
+#align apply_wcovby_apply_iff apply_wcovby_apply_iff
+
+@[simp]
+theorem to_dual_wcovby_to_dual_iff : toDual b ⩿ toDual a ↔ a ⩿ b :=
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:139]:
```suggestion
theorem toDual_wcovby_toDual_iff : toDual b ⩿ toDual a ↔ a ⩿ b :=
```

File: Mathlib/Order/Cover.lean [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+  and_congr_right' <| forall_congr' fun _ => forall_swap
+#align to_dual_wcovby_to_dual_iff to_dual_wcovby_to_dual_iff
+
+@[simp]
+theorem of_dual_wcovby_of_dual_iff {a b : αᵒᵈ} : ofDual a ⩿ ofDual b ↔ b ⩿ a :=
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:144]:
```suggestion
theorem ofDual_wcovby_ofDual_iff {a b : αᵒᵈ} : ofDual a ⩿ ofDual b ↔ b ⩿ a :=
```

File: Mathlib/Order/Cover.lean [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+  and_congr_right' <| forall_congr' fun _ => forall_swap
+#align of_dual_wcovby_of_dual_iff of_dual_wcovby_of_dual_iff
+
+alias to_dual_wcovby_to_dual_iff ↔ _ Wcovby.to_dual
+
+alias of_dual_wcovby_of_dual_iff ↔ _ Wcovby.of_dual
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:150]: These should be changed as well.

File: Mathlib/Order/Cover.lean [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+  h.2 hc.1 hc.2
+#align not_covby not_covby
+
+theorem densely_ordered_iff_forall_not_covby : DenselyOrdered α ↔ ∀ a b : α, ¬a ⋖ b :=
+  ⟨fun h _ _ => @not_covby _ _ _ _ h, fun h =>
+    ⟨fun _ _ hab => exists_lt_lt_of_not_covby hab <| h _ _⟩⟩
+#align densely_ordered_iff_forall_not_covby densely_ordered_iff_forall_not_covby
+
+@[simp]
+theorem to_dual_covby_to_dual_iff : toDual b ⋖ toDual a ↔ a ⋖ b :=
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:250]:
```suggestion
theorem toDual_covby_toDual_iff : toDual b ⋖ toDual a ↔ a ⋖ b :=
```

File: Mathlib/Order/Cover.lean [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+  and_congr_right' <| forall_congr' fun _ => forall_swap
+#align to_dual_covby_to_dual_iff to_dual_covby_to_dual_iff
+
+@[simp]
+theorem of_dual_covby_of_dual_iff {a b : αᵒᵈ} : ofDual a ⋖ ofDual b ↔ b ⋖ a :=
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:255]:
```suggestion
theorem ofDual_covby_ofDual_iff {a b : αᵒᵈ} : ofDual a ⋖ ofDual b ↔ b ⋖ a :=
```

File: Mathlib/Order/Cover.lean [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+  and_congr_right' <| forall_congr' fun _ => forall_swap
+#align of_dual_covby_of_dual_iff of_dual_covby_of_dual_iff
+
+alias to_dual_covby_to_dual_iff ↔ _ Covby.to_dual
+
+alias of_dual_covby_of_dual_iff ↔ _ Covby.of_dual
+
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:262]: Here as well

File: Mathlib/Order/Cover.lean [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+theorem AntisymmRel.trans_covby (hab : AntisymmRel (· ≤ ·) a b) (hbc : b ⋖ c) : a ⋖ c :=
+  ⟨hab.1.trans_lt hbc.lt, fun _ had hdc => hbc.2 (hab.2.trans_lt had) hdc⟩
+#align antisymm_rel.trans_covby AntisymmRel.trans_covby
+
+theorem covby_congr_left (hab : AntisymmRel (· ≤ ·) a b) : a ⋖ c ↔ b ⋖ c :=
+  ⟨hab.symm.trans_covby, hab.trans_covby⟩
+#align covby_congr_left covby_congr_left
+
+theorem Covby.trans_antisymm_rel (hab : a ⋖ b) (hbc : AntisymmRel (· ≤ ·) b c) : a ⋖ c :=
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:318]:
```suggestion
theorem Covby.trans_antisymmRel (hab : a ⋖ b) (hbc : AntisymmRel (· ≤ ·) b c) : a ⋖ c :=
```

File: Mathlib/Order/Cover.lean [hunk 9]
```diff
@@ -0,0 +1,554 @@
...
+
+theorem covby_congr_right (hab : AntisymmRel (· ≤ ·) a b) : c ⋖ a ↔ c ⋖ b :=
+  ⟨fun h => h.trans_antisymm_rel hab, fun h => h.trans_antisymm_rel hab.symm⟩
+#align covby_congr_right covby_congr_right
+
+instance : IsNonstrictStrictOrder α (· ⩿ ·) (· ⋖ ·) :=
+  ⟨fun _ _ =>
+    covby_iff_wcovby_and_not_le.trans <| and_congr_right fun h => h.wcovby_iff_le.not.symm⟩
+
+instance Covby.is_irrefl : IsIrrefl α (· ⋖ ·) :=
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:330]:
```suggestion
instance Covby.isIrrefl : IsIrrefl α (· ⋖ ·) :=
```

File: Mathlib/Order/Cover.lean [hunk 10]
```diff
@@ -0,0 +1,554 @@
...
+
+variable [PartialOrder α] {a b c : α}
+
+theorem Wcovby.eq_or_eq (h : a ⩿ b) (h2 : a ≤ c) (h3 : c ≤ b) : c = a ∨ c = b := by
+  rcases h2.eq_or_lt with (h2 | h2); · exact Or.inl h2.symm
+  rcases h3.eq_or_lt with (h3 | h3); · exact Or.inr h3
+  exact (h.2 h2 h3).elim
+#align wcovby.eq_or_eq Wcovby.eq_or_eq
+
+/-- An `iff` version of `wcovby.eq_or_eq` and `wcovby_of_eq_or_eq`. -/
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:164]:
```suggestion
/-- An `iff` version of `Wcovby.eq_or_eq` and `wcovby_of_eq_or_eq`. -/
```

File: Mathlib/Order/Cover.lean [hunk 11]
```diff
@@ -0,0 +1,554 @@
...
+section LT
+
+variable [LT α] {a b : α}
+
+/-- `Covby a b` means that `b` covers `a`: `a < b` and there is no element in between. -/
+def Covby (a b : α) : Prop :=
+  a < b ∧ ∀ ⦃c⦄, a < c → ¬c < b
+#align covby Covby
+
+-- mathport name: «expr ⋖ »
```
Comment (dupuisf) [inline Mathlib/Order/Cover.lean:221]:
```suggestion
```

**Review by dupuisf — COMMENTED** [review]
Just a few name changes, otherwise looks good!

bors d+

**Comment by riccardobrasca** [comment]
bors merge

**Pull request closed without merging**